### PR TITLE
Refactor chat handoff

### PR DESCRIPTION
**Pull request by bsadecki-magi**
Here's a PR description for the chat handoff refactoring:

# Description

This PR introduces a refactor of the chat handoff system to improve maintainability and extensibility.

## Key Changes
- Created a new `HandoffStrategy` interface to standardize handoff implementations
- Added dedicated strategy classes for Zendesk and Front integrations
- Centralized message formatting and event handling logic

## Breaking Changes

None. This refactoring maintains backward compatibility while providing a more robust foundation for future handoff integrations.

## Related Issues

- Sets up salesforce handoff strategies
- Addresses technical debt in chat handoff system
- Improves maintainability for future integrations

## Checklist

- [x] Added tests
- [x] Maintained backward compatibility
- [x] Tested edge cases
- [x] Test ZD and Front handoff locally

### CODE REVIEW

File: lib/handoff/FrontStrategy.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import type { HandoffStrategy } from "./HandoffStrategy";
+import type { Message, HandoffChatMessage } from "@/types";
+import type { Front } from "@/types/front";
+import { isChatUserMessage, isBotMessage } from "@/types";
+
```
Comment (bsy) [inline lib/handoff/FrontStrategy.ts:5]: const MESSAGE_TYPES = ["USER", "bot"] as const

File: lib/handoff/FrontStrategy.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+export class FrontStrategy implements HandoffStrategy {
+  getMessagesEndpoint = "/api/front/messages";
+  getConversationsEndpoint = "/api/front/conversations";
+
+  formatMessages(
+    messages: Message[],
+    mavenConversationId: string,
+  ): HandoffChatMessage[] {
+    return messages
+      .filter((message) => ["USER", "bot"].includes(message.type))
```
Comment (bsy) [inline lib/handoff/FrontStrategy.ts:15]:
```suggestion
      .filter((message) => MESSAGE_TYPES.includes(message.type))
```

File: lib/handoff/HandoffStrategy.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+export interface HandoffStrategy {
+  formatMessages: (
+    messages: Message[],
+    mavenConversationId: string,
+  ) => HandoffChatMessage[];
+  handleChatEvent: (event: any) => {
+    agentName: string | null;
+    formattedEvent: any;
+  };
+  getMessagesEndpoint: string;
```
Comment (bsy) [inline lib/handoff/HandoffStrategy.ts:12]: For immutability:
```
readonly getMessagesEndpoint: string;
readonly getConversationsEndpoint: string;
 ```

**Review by bsy — APPROVED** [review]
lgtm, just some minor suggestions for typing and immutability

File: lib/handoff/HandoffStrategy.ts (hunk 3 above)
Comment (bsadecki-magi) [reply lib/handoff/HandoffStrategy.ts:12]: Incorporated in next PR, thx!
